src/netlink.c

```diff
@@ -0,0 +1,431 @@
+		g_debug("mnl_socket_sendto %s", strerror(errno));
+		goto out;
+	}
+
+	ret = mnl_socket_recvfrom(hndl->nl, buf, sizeof(buf));
+	if (ret == -1) {
+		g_debug("mnl_socket_recvfrom failed %s", strerror(errno));
+		goto out;
+	}
+
+	while (ret > 0) {
+		ret = mnl_cb_run(buf, (size_t)ret, seq, portid, NULL, NULL);
+		if (ret <= 0) {
+			break;
+		}
+		ret = mnl_socket_recvfrom(hndl->nl, buf, sizeof(buf));
+	}
+
+	if (ret == -1) {
+		g_debug("netlink error");
+		goto out;
+	}
+
+	status = true;
+out:
+	return status;
+}
+
+/*!
+ * Netlink command equivalent to
+ * ip link set dev <interface> <up|down> 
+ *
+ * \param handle returned from a call to netlink_init
+ * \param interface device name to enable\disable
+ * \param enable if true device will enabled, else disabled
+ *
+ * \return \c true on success else \c false
+ */
+gboolean
+netlink_link_enable(struct netlink_handle *const hndl,
+		    const gchar *const interface, gboolean enable)  {
+	char buf[MNL_SOCKET_BUFFER_SIZE];
+	struct nlmsghdr *nlh;
+	struct ifinfomsg *ifm;
+	guint change = 0, flags = 0;
+
+	g_debug("netlink_link_enable %s", interface);
+
+	if (enable) {
+		change |= IFF_UP;
+		flags |= IFF_UP;
+	} else {
+		change |= IFF_UP;
+		flags &= (guint)~IFF_UP;
+	}
+
+	nlh = mnl_nlmsg_put_header(buf);
+	nlh->nlmsg_type = RTM_NEWLINK;
+	nlh->nlmsg_flags = NLM_F_REQUEST | NLM_F_ACK;
+	ifm = mnl_nlmsg_put_extra_header(nlh, sizeof(*ifm));
+	ifm->ifi_family = AF_UNSPEC;
+	ifm->ifi_change = change;
+	ifm->ifi_flags = flags;
+
+	mnl_attr_put_str(nlh, IFLA_IFNAME, interface);
+
+	return netlink_execute(hndl, nlh);
+}
+
+/*!
+ * Netlink command equivalent to
+ * ip link add name ${bridge name} type bridge
+ *
+ * \param handle returned from a call to netlink_init
+ * \param name of the bridge to create
+ *
+ * \return \c true on success else \c false
+ */
+gboolean
+netlink_link_add_bridge(struct netlink_handle *const hndl,
+			const gchar *const name)  {
+	char buf[MNL_SOCKET_BUFFER_SIZE];
+	struct nlmsghdr *nlh;
+	struct ifinfomsg *ifm;
+	struct nlattr* link_attr;
+
+	g_debug("netlink_link_add_bridge %s", name);
+
+	nlh = mnl_nlmsg_put_header(buf);
+	nlh->nlmsg_type = RTM_NEWLINK;
+	nlh->nlmsg_flags = NLM_F_REQUEST | NLM_F_CREATE | NLM_F_EXCL | NLM_F_ACK;
+	ifm = mnl_nlmsg_put_extra_header(nlh, sizeof(*ifm));
+	ifm->ifi_family = AF_UNSPEC;
+
+	mnl_attr_put_str(nlh, IFLA_IFNAME, name);
+	link_attr = mnl_attr_nest_start(nlh, IFLA_LINKINFO);
+	mnl_attr_put_str(nlh, IFLA_INFO_KIND, "bridge");
+	mnl_attr_nest_end(nlh, link_attr);
+
+	return netlink_execute(hndl, nlh);
+}
+
+/*!
+ * Netlink command equivalent to
+ * ip link set dev ${interface name} master ${bridge name}
+ *
+ * \param handle returned from a call to netlink_init
+ * \param dev index of the device to add to the bridge
+ * \param master index of the bridge
+ *
+ * \return \c true on success else \c false
+ */
+gboolean
+netlink_link_set_master(struct netlink_handle *const hndl,
```

Comment:
@jodh-intel fixed... dev and master can have any value. the netlink call internally will validate if index supplied is valid.